code/code/sys/indexdata.cc

```diff
@@ -18,167 +18,108 @@ extern FILE *mob_f;
+      numberLoad(0) {}
+
+mobIndexData::mobIndexData(int _virt, const sstring &_name, const sstring &_short_desc,
+                           const sstring &_long_desc, const sstring &_description, int _max_exist,
+                           int _spec, float _weight, long faction, long Class, long level,
+                           long race)
+    : faction(faction), Class(Class), level(level), race(race) {
+  virt = _virt;
+  name = _name;
+  short_desc = _short_desc;
+  long_desc = _long_desc;
+  description = _description;
+  max_exist = _max_exist;
+  spec = _spec;
+  weight = _weight;
 }
 
-objIndexData::objIndexData() :
-  ex_description(NULL),
-  max_struct(-99),
-  armor(-99),
-  where_worn(0),
-  itemtype(MAX_OBJ_TYPES),
-  value(-99)
-{
-}
-
-objIndexData & objIndexData::operator= (const objIndexData &a)
-{
-  if (this == &a)
-    return *this;
-
-  indexData::operator=(a);
-
-  // use copy operator;
-  if (a.ex_description)
-    ex_description = new extraDescription(*a.ex_description);
-  else
-    ex_description = NULL;
-
-  max_struct = a.max_struct;
-  armor = a.armor;
-  where_worn = a.where_worn;
-  itemtype = a.itemtype;
-  value = a.value;
-
-  int i;
-  for(i=0;i<MAX_OBJ_AFFECT;++i){
-    affected[i]=a.affected[i];
-  }
-
-  return *this;
+mobIndexData::mobIndexData(const mobIndexData &a)
+    : indexData(a),
+      faction(a.faction),
+      Class(a.Class),
+      level(a.level),
+      race(a.race),
+      doesLoad(a.doesLoad),
+      numberLoad(a.numberLoad) {}
+
+objIndexData::objIndexData()
+    : max_struct(-99),
+      armor(-99),
+      where_worn(0),
+      itemtype(MAX_OBJ_TYPES),
+      value(-99),
+      ex_description(nullptr) {}
+
+objIndexData::objIndexData(int _virt, sstring _name, sstring _short_desc, sstring _long_desc,
+                           int _max_exist, int _spec, float _weight, byte max_struct,
+                           unsigned int where_worn, ubyte itemtype, int value, sstring _description,
+                           extraDescription *ex_description, objAffData _affected[MAX_OBJ_AFFECT])
+    : max_struct(max_struct), where_worn(where_worn), itemtype(itemtype), value(value), ex_description(ex_description) {
+  virt = _virt;
+  name = _name;
+  short_desc = _short_desc;
+  long_desc = _long_desc;
+  max_exist = _max_exist;
+  spec = _spec;
+  weight = _weight;
+  description = _description;
+  
+  for (int i = 0; i < MAX_OBJ_AFFECT; i++)
+    affected[i] = _affected[i];
 }
 
-objIndexData::objIndexData(const objIndexData &a) :
-  indexData(a),
-  max_struct(a.max_struct),
-  armor(a.armor),
-  where_worn(a.where_worn),
-  itemtype(a.itemtype),
-  value(a.value)
-{
+objIndexData::objIndexData(const objIndexData &a)
+    : indexData(a),
+      max_struct(a.max_struct),
+      armor(a.armor),
+      where_worn(a.where_worn),
+      itemtype(a.itemtype),
+      value(a.value) {
   // use copy operator;
   if (a.ex_description)
     ex_description = new extraDescription(*a.ex_description);
```

Comment:
Is it leaking memory?

Reply:
Perhaps we could stringify it

Reply:
Should be resolved in https://github.com/sneezymud/sneezymud/pull/380/commits/ca9f8b35b90fb1f6818a43d8dc48feadd8831a56